### DISCUSSION TITLE
Packaging a Restlet app into a native binary image

### DISCUSSION THREAD

**Discussion by kerbymart** [restlet/restlet-framework-java#1412] [Q&A]
Hello there! I wanted to let you know that I have created a simple POC for a native image that uses Restlet resources. You can find it on GitHub here: [github.com/kerbymart/restlet-native-image](https://github.com/kerbymart/restlet-native-image). 

Please note that I'm still working on writing the README, but I have managed to build the binary successfully. However, when I tried to access a resource with `http://localhost:8080/hello`, which should be available, I encountered a 405 error. Interestingly, the uber jar works fine, but for the "native image", I'm not sure what is causing the route to not be mapped. The build process had no errors.

**Comment by kerbymart** [comment]
Just run `mvn clean package` and it should build the `example` binary.

**Comment by jlouvel** [comment]
@kerbymart That's really cool you are exploring this. This is something I'd like to officially support in Restlet 3.0. I don't have time yet to play with your code and help debug, but once 2.5. is more advanced, I'd love to explore this PoC further.
